### PR TITLE
Added access token resource

### PR DESCRIPTION
**Pull request by donovanmuller**
Adds [Access Token](https://www.jfrog.com/confluence/display/JFROG/Access+Tokens) support to the provider.
Currently only the ability to create (create and update) and revoke (delete) a token is supported.

There is no datasource support currently because of current limitations of the [Artifactory REST API](https://www.jfrog.com/confluence/display/JFROG/Artifactory+REST+API). 
When [creating](https://www.jfrog.com/confluence/display/JFROG/Artifactory+REST+API#ArtifactoryRESTAPI-CreateToken) a new token, the response would look like this:

```json
{
   "access_token":   "<the access token>",
   "expires_in":    "<Validity period in seconds>",
   "scope":         "<access scope>",
   "token_type":    "Bearer",
   "refresh_token": "<the refresh token if access_token is refreshable>"
}
```

and the only available REST resource available to read tokens is to get a [list of tokens](https://www.jfrog.com/confluence/display/JFROG/Artifactory+REST+API#ArtifactoryRESTAPI-GetTokens), which returns a response like this:

```jsonc
{
  "tokens" : [ {
    "issuer" : "jfrt@01ejth5fa032aa1x9197451qk3",
    "subject" : "jfrt@01ejth5fa032aa1x9197451qk3/users/abdma4q-admin",
    "expiry" : 1605794711,
    "refreshable" : false,
    "token_id" : "d61a6655-9169-4236-84c6-6e836b0bac01",
    "issued_at" : 1603375511
  }, {
...
}
```

Unfortunately, there is no obvious way to correlate a token from the list of tokens returned from `GET /api/security/token` to the token that was just created with `POST /api/security/token`. I.e. There are no uniquely identifiable fields that match in either reponse.

### CODE REVIEW

**Review by k0da — APPROVED** [review]
LGTM

**Review by kuritka — APPROVED** [review]
I checked coding-style related to rest of the project  (filenames, syntax, test style) and run analysis tools. LGTM

File: pkg/artifactory/resource_artifactory_access_token.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+	accessTokenOption.Refreshable = artifactory.String(strconv.FormatBool(*refreshable))
+	accessTokenOption.Audience = d.getStringRef("audience", false)
+
+	return accessTokenOption, nil
+}
+
+func resourceAccessTokenCreate(d *schema.ResourceData, m interface{}) error {
+	c := m.(*ArtClient).ArtOld
+
+	accessTokenOption, err := unmarshalToken(d)
```
Comment (kuritka) [inline pkg/artifactory/resource_artifactory_access_token.go]: unhandled error, or better don't return err (`unmarshalToken` returns always nil)

**Review by kuritka — CHANGES_REQUESTED** [review]

File: pkg/artifactory/resource_artifactory_access_token.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+}
+
+func resourceAccessTokenRead(d *schema.ResourceData, m interface{}) error {
+	return nil
+}
+
+func resourceAccessTokenDelete(d *schema.ResourceData, m interface{}) error {
+	c := m.(*ArtClient).ArtOld
+
+	_, resp, err := c.V1.Security.RevokeToken(context.Background(), v1.AccessTokenRevokeOptions{
```
Comment (kuritka) [inline pkg/artifactory/resource_artifactory_access_token.go:98]: Unhandled error,  `resp` is `*string` and can be nil, so if error occurs `resp.StatusCode` possibly fails on nil reference

File: pkg/artifactory/resource_artifactory_access_token.go (hunk 1 above)
Comment (donovanmuller) [reply pkg/artifactory/resource_artifactory_access_token.go]: @kuritka this method matches the Terraform contract defined for [`CreateFunc`](https://github.com/hashicorp/terraform/blob/ce9e4b2ebdd68609b2e9d34866bafb1c3c773109/helper/schema/resource.go#L199).

Comment (kuritka) [reply pkg/artifactory/resource_artifactory_access_token.go]: @donovanmuller sorry for the previous comment, I was thinking `unmarshalToken` function which returns always nil, so we can remove returned error type. 

line 77 would look like 
```
accessTokenOption := unmarshalToken(d) 
```
The function contract you mentioned is related to parent `resourceAccessTokenCreate`

Comment (donovanmuller) [reply pkg/artifactory/resource_artifactory_access_token.go]: @kuritka got you, have adjusted.

**Review by k0da — APPROVED** [review]

**Review by kuritka — APPROVED** [review]
LGTM 👍

**Pull request merged**